Shopify/cli
# Disable dev preview on exit

### WHY are these changes introduced?


Extracted disable dev preview when dev command finishes



### WHAT is this pull request doing?
- Disable developer preview when `dev` commands finishes
-  Modify `ConcurrentOutput` component to finish the command (press `q` or `ctrl + c` using the `AbortController` instead of `treeKill`
- `dev` add a graceful finishing process to close every open external process. 
  - Include a timeout to force the finishing of the command
  - dev analytics is sent when the command is finished



### How to test your changes?
- Generate a Remix app with `theme app extension`, `checkout UI extension` and one `function`
- Run the `dev` command with the `--verbose` flag
- Check in the partners dashboard that the dev preview is enabled
- Press `q` or `CTRL+C` and check debug messages about the different child processes being stopped
- Check in the partners dashboard that the dev preview is disabled


### Post-release steps



### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
